core/src/trezor/strings.py

```diff
@@ -64,3 +70,57 @@ def format_duration_ms(milliseconds: int) -> str:
         divisor = 1
 
     return format_plural("{count} {plural}", milliseconds // divisor, unit)
+
+
+def _calculate_timestamp_correction(timestamp: int) -> (Tuple[tuple, int]):
+    """
+    utime module can't convert timestamp to datetime with seconds precision.
+    returns date in tuple format and correction in seconds
+    from the utime-calculated date's midnight
+    to the correct datetime
+    Example:
+        timestamp                   1616057224
+        correct datetime            "2021-03-18 08:47:04"
+        utime-calculated datetime   "2021-03-18 08:46:56"
+        midnight datetime           "2021-03-18 00:00:00"
+        returned date               (2021, 3, 18, 0, 0, 0, 3, 77, 0)
+        correction                  31624 (~ 8:47:04)
+    """
+    approx_datetime = utime.localtime(timestamp)
```

Comment:
On the actual device this fails with:
```
AttributeError: 'module' object has no attribute 'localtime'
```

Reply:
Even if we got this working, I suspect that it might give an incorrect time on the device, because the STM32 implementation seems to be using 2000-01-01 as the beginning of the epoch as opposed to the UNIX implementation which uses 1970-01-01. It's hard to believe, but that's what I see in the C code. I didn't actually test.

Reply:
Yes, from my limited sources, I expect the behavior you described. Sorry for not mentioning it in the PR description.

Reply:
If the two implementations are really using different epochs, then I have doubts whether we should be using `utime` at all. We would then have to have a piece of code like "If not EMULATOR, then subtract 30 years", which just seems very wrong.

Secondly, there is https://github.com/trezor/trezor-firmware/issues/741, which indicates that there may have been some opposition to adding the utime module to the firmware build.

Reply:
I just added PR https://github.com/trezor/trezor-firmware/pull/1548 to update uPy to 1.14 - among other things it introduces `utime.gmtime()` function which is exactly what we need here I suppose.

Reply:
fwiw #741 is not opposed to adding utime, but adding full python's `datetime`. At that time there was probably no date processing in `utime`, but now there is.